js/main.js

```diff
@@ -206,6 +221,13 @@ function validate_xrp_secret(){
   var xrp_secret = document.getElementById("xrp_secret");
 
   inputs_valid.xrp_secret = offline_api.isValidSecret(xrp_secret.value);
+  if (!inputs_valid.xrp_secret) {
+      // not a valid secret, so check to see if it's a valid HEX private key instead.
+      // Is it > 64 chars?  if so, could be padded with double-zero at start
+      const xrp_privatekey = (xrp_secret.length > 64) ? xrp_secret.value.slice(2) : xrp_secret.value;
```

Comment:
Do you know any wallets that generate 64 character keys?

If not lets assume that private keys are 66 chars as ripple-lib generates these as such, as these first two chars will be used to distinguish between secp256k1 and ED25519 keys below.

So we can remove this xrp_privatekey const here and just change the following to:

var re = /^[A-Fa-f0-9]{**66**}/g;                                                                                                                      
inputs_valid.xrp_secret = **xrp_secret.value.length == 66 &&** re.test(**xrp_secret.value**);

Reply:
Hi @movitto.  Reason for checking for 64 length is that tools like Ian Colemans BIP39 tool will generate addresses with private keys at hex length 64 (https://iancoleman.io/bip39/)  So I think we need to keep that in (if you agree?)

I'll add the ED check too, my bad.  Standby for corrections.